src/java/org/apache/cassandra/db/memtable/TrieMemtable.java

```diff
@@ -360,8 +360,8 @@ public FlushablePartitionSet<MemtablePartition> getFlushSet(PartitionPosition fr
     {
         boolean allPositionsToFlush = from == null && to == null
                                       || from != null && to != null
-                                         && from.isMinimum()
-                                         && to.isMaximum();
+                                         && from.isMinimum() && from.kind() == PartitionPosition.Kind.MIN_BOUND
```

Comment:
The minimum token is something reserved for boundaries (i.e. the computed hash is adjusted to avoid it) thus the `MIN_BOUND` check here is unnecessary. AFAIAA other uses of `isMinimum` don't check this, and I believe we also use minimum with `MAX_BOUND` in some cases, which will fail this check when the intention is to fully cover the lower side of the ring.